lib/trainman/build.js

```diff
@@ -134,10 +137,40 @@ module.exports = function( plugins, config, options ) {
         version: 'Using "version" here has been deprecated. Please use "buildSignature" instead.'
       };
 
-      return queueStreams( sources ).
-        pipe( plugins.template( locals ) ).
-        pipe( plugins.gulp.dest( config.publicDir ) ).
-        pipe( plugins.connect.reload() );
+      if ( config.environment.locals ) {
+        plugins.extend( true, locals, config.environment.locals );
+      }
+
+      // Template Filters
+      var allButIndex = filter(['**', '!app/templates/index.html'], {restore: true});
+      var index = filter(['*', 'app/templates/index.html'], {restore: true});
+
+      var mainStream = queueStreams( sources ).
+                          pipe( plugins.template( locals ) );
+
+      var indexStream = mainStream.
+                          pipe( index ).
+                          pipe( plugins.gulp.dest( config.publicDir ) ).
+                          pipe( print( function(filepath) {
+                            return "index path: " + filepath;
+                          }));
+                          // pipe( plugins.connect.reload() );
+
+      var otherStreams = mainStream.
+                          pipe( allButIndex ).
+                          pipe( plugins.gulp.dest( config.publicDir + '/assets/templates' ) ).
```

Comment:
Do you think you could use assetOutputPath + source.base or something like that?